### PR TITLE
PLAN-1697: Expander sections for filter dropdown

### PR DESCRIPTION
**Pull request by LALuis**
The idea is to keep the component as it is and when a category and its options array is provided we are displaying the expander sections for each category.

Jira: https://sig-gis.atlassian.net/browse/PLAN-1697

Design: https://www.figma.com/design/njHEIn6MWPYJ5hw4we0lFY/Impacts?node-id=1760-51648&node-type=instance&m=dev

Result:
image

### CODE REVIEW

File: src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.ts [hunk 1]
```diff
@@ -89,6 +92,7 @@ export class FilterDropdownComponent<T> implements OnInit {
   private previousSelections: T[] = [];
 
   ngOnInit(): void {
+    this.displayCategories = this.menuItems.some((e) => (e as any).category);
```
Comment (lastminutediorama) [inline src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.ts:95]: I like that this can be set dynamically, thanks.

**Review by lastminutediorama — APPROVED** [review]

File: src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.html [hunk 2]
```diff
@@ -46,20 +46,50 @@
...
-          [color]="'primary'"
-          [checked]="isInSelection(item)"></mat-checkbox>
+    <ng-container
+      *ngIf="
+        displayCategories && displayedItems && displayedItems.length > 0;
+        else noCategory
+      ">
+      <div
+        *ngFor="let categoryGroup of displayedItems"
+        (click)="$event.stopPropagation()">
```
Comment (pflopez) [inline src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.html:56]: Not sure I understand why we are adding this here, a `div` does not have a default click action.
Should this be on the `button` instead?

**Review by pflopez — APPROVED** [review]
Looking good, just a comment on the div click thing, but not a blocker

File: src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.html (hunk 2 above)
Comment (LALuis) [reply src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.html]: Updated, I added it to the expander. Thank you Pablo!

**Pull request merged**